readme.md

```diff
@@ -1,161 +1,181 @@
+#### How to use the generated code
 
-p := &channel.ReceiveLightMeasurementParams{}
-err := p.Parse("smartylighting/streetlights/1/0/event/(.+)/lighting/measured")
-// Populates p with channel parameters
-// p = { StreetlightId: "<my-id>" }
-```
+The above code currently generates a Go module that has a AMQP subscriber.
 
-## Getting Started
-
-### Dependencies:
-The AsyncAPI Generation tool is required to execute this template. Find it here : https://github.com/asyncapi/generator/#install-the-cli.
-
-1. Install AsyncAPI Generation tool
-	```bash
-	$ npm install -g @asyncapi/generator
-	```
-2. Clone the template to wherever you like
-	```bash
-	$ cd ~/dev/tools/
-	$ git clone https://github.com/asyncapi/go-template.git
-	```
-3. Run the generator.  
-	Arguments:
-	* The path to your Async specification file
-	* The path to the template folder of this repo
-	* Destination directory for output
-	```
-	$ ag ./asyncapi.yaml ~/dev/tools/go-template/template -o ./<dest-dir>
-	```
-
-## What does the generation look like? 
-Examples based on the Streetlights example specification
-
-#### Message Sample
-```golang
-//   Location: message/<msg_name/id>.go
-
-
-// DimLight is used to command a particular streetlight to dim the lights.
-type DimLight struct {
-	transport.Message
-
-	// ContentType indicates the specified MIME type for this message. If empty, the defaultContentType should be used 
-	ContentType string
-	
-	// Payload contains the content defined by the payload AsyncAPI field
-	Payload model.DimLightPayload
-}
-```
-#### Schema Sample
-```golang
-//   Location: model/<schema_name>.go
+##### Pre-requisites
+To run the generated code the following needs to be installed
+
+1. go 1.16 +
+2. rabbitmq-server OR docker
 
-// DimLightPayload is a Schema defined in the AsyncAPI specification
-type DimLightPayload struct {
-    // Percentage is a property defined in the AsyncAPI specification
-    Percentage *int64 `json:"percentage"`
+##### Running the code
+
+1. Navigate to the path where the code was generated
+2. Run the following commands to download the dependencies
+```bash
+go mod download
+go mod tidy
+```
+3. Currently the code does not utilize the server bindings to generate the server URI. It is currently hardcoded to point to a local instance of `rabbitmq`. It is hardcoded as `"amqp://guest:guest@localhost:5672/"` at `<generated-code>/config/server.go`. Change it as per your rabbitmq instance requirements
+4. Finally to execute the code run
+```bash
+go run main.go
+```
+5. Running local instance of `rabbitmq`, navigate to it using `http://localhost:15672/` with username and password `guest`/ `guest` (These are default rabbitmq credentials). 
+   FYI one can start an instance of `rabbitmq` using  `docker` as follow
+   ```
+   docker run -d -p 15672:15672 -p 5672:5672 rabbitmq:3-management
+   ```
+6. Create a queue as per the async api spec
+7. Publish a message to the queue as per the async api spec
```

Comment:
```suggestion
7. Publish a message to the queue as per the AsyncAPI file
```

Reply:
honestly would be cool to have at least a link to doc that explains how to do it, or simple curl, or something

Reply:
curl is such a good idea I am going to add that